coreneuron/io/reports/report_handler.cpp

```diff
@@ -0,0 +1,230 @@
+            case CompartmentReport:
+                vars_to_report =
+                    get_compartment_vars_to_report(nt, m_report_config.target, nt._actual_v);
+                register_compartment_report(nt, m_report_config, vars_to_report);
+                break;
+            case IMembraneReport:
+                vars_to_report = get_compartment_vars_to_report(nt, m_report_config.target,
+                                                                nt.nrn_fast_imem->nrn_sav_rhs);
+                register_compartment_report(nt, m_report_config, vars_to_report);
+                break;
+            default:
+                vars_to_report = get_custom_vars_to_report(nt, m_report_config, nodes_to_gid);
+                register_custom_report(nt, m_report_config, vars_to_report);
+        }
+        if (!vars_to_report.empty()) {
+            std::unique_ptr<ReportEvent> report_event =
+                std::make_unique<ReportEvent>(dt, t, vars_to_report, m_report_config.output_path);
+            report_event->send(t, net_cvode_instance, &nt);
+            m_report_events.push_back(std::move(report_event));
+        }
+    }
+#else
+    if (nrnmpi_myid == 0) {
+        std::cerr << "[WARNING] : Can't enable reports, recompile with ReportingLib! \n";
+    }
+#endif  // defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+}
+
+#if defined(ENABLE_REPORTINGLIB) || defined(ENABLE_SONATA_REPORTS)
+void ReportHandler::register_soma_report(const NrnThread& nt,
+                                         ReportConfiguration& config,
+                                         const VarsToReport& vars_to_report) {
+    if (nrnmpi_myid == 0) {
+        std::cerr << "[WARNING] : Report '" << config.output_path << "' not supported!\n";
+    }
+}
+void ReportHandler::register_compartment_report(const NrnThread& nt,
+                                                ReportConfiguration& config,
+                                                const VarsToReport& vars_to_report) {
+    if (nrnmpi_myid == 0) {
+        std::cerr << "[WARNING] : Report '" << config.output_path << "' not supported!\n";
+    }
+}
+void ReportHandler::register_custom_report(const NrnThread& nt,
+                                           ReportConfiguration& config,
+                                           const VarsToReport& vars_to_report) {
+    if (nrnmpi_myid == 0) {
+        std::cerr << "[WARNING] : Report '" << config.output_path << "' not supported!\n";
+    }
+}
+VarsToReport ReportHandler::get_soma_vars_to_report(const NrnThread& nt,
+                                                    const std::set<int>& target,
+                                                    double* report_variable) const {
+    VarsToReport vars_to_report;
+    const auto* mapinfo = static_cast<NrnThreadMappingInfo*>(nt.mapping);
+    if (!mapinfo) {
+        std::cerr << "[SOMA] Error : mapping information is missing for a Cell group " << nt.ncell
+                  << "\n";
```

Comment:
```suggestion
                  << '\n';
```
I know the process is about to exit, but still :)